src/main/kotlin/org/jmailen/gradle/kotlinter/KotlinterPlugin.kt

```diff
@@ -93,6 +111,31 @@ class KotlinterPlugin : Plugin<Project> {
             it.group = "build setup"
             it.description = "Installs Kotlinter Git pre-commit hook"
         }
+
+    private fun Project.createKtlintConfiguration(): Configuration = configurations.maybeCreate(KTLINT_CONFIGURATION_NAME).apply {
+        isCanBeResolved = true
+        isCanBeConsumed = false
+        isVisible = false
+
+        val dependencyProvider = provider {
+            this@createKtlintConfiguration.dependencies.create(
+                "com.pinterest:ktlint:${Versions.ktlint}"
```

Comment:
Any way to reference a BOM or something like that?

Reply:
The reason why I ask is that we include 8 "library" type dependencies from ktlint normally in the plugin and do not include the cli part of ktlint. It would be preferable to keep doing that as to not bring in unnecessary code.

With this method of inclusion, wouldn't we also make the ktlint dependencies in `build.gradle.kts` `compileOnly`?